subscription/views.py

```diff
@@ -1016,39 +1026,75 @@ def post(self, request):
 #                 return premium_page.units_consumed
 #     return 1  # Default to 1 unit if not found in PremiumUsageUnits
 
-def get_premium_units(url):
-    logger.debug("get_premium_units " + url)
+# def get_usage_units(url):
+    # logger.debug("get_premium_units " + url)
+    # try:
+    #     # First, try an exact match
+    #     premium_page = PremiumUsageUnits.objects.get(url=url)
+    #     return premium_page.units_consumed
+    # except PremiumUsageUnits.DoesNotExist:
+    #     # If no exact match, try containment check
+    #     for premium_page in PremiumUsageUnits.objects.all():
+    #         if premium_page.url in url:
+    #             return premium_page.units_consumed
+        
+    #     # If still no match, try regex matching as a fallback
+    #     for premium_page in PremiumUsageUnits.objects.all():
+    #         if re.match(premium_page.url, url):
+    #             return premium_page.units_consumed
+    
+    # return 1  # Default to 1 unit if not found in PremiumUsageUnits
+
+def get_usage_units(url):
+    logger.debug("get_usage_units_from_uri_pattern: %s" % url)
+
+    uri_pattern_object = None
+
+    # Step 1: Attempt to find a match in UriPattern table using regex matching
+    for pattern_entry in UriPattern.objects.all():
```

Comment:
You may want to add a filter condition to only get the "Paid" type of UriPattern, there are also "Login required" patterns which could be mistakenly recognized as paid pattern